eras/conway/impl/test/Test/Cardano/Ledger/Conway/GovActionReorderSpec.hs

```diff
@@ -10,31 +11,25 @@ import Data.Foldable (Foldable (..))
 import Data.List (sort, sortOn)
 import qualified Data.Sequence.Strict as Seq
 import Test.Cardano.Ledger.Binary.Arbitrary ()
-import Test.Cardano.Ledger.Common (Spec, describe, prop, shuffle)
+import Test.Cardano.Ledger.Common
 import Test.Cardano.Ledger.Conway.Arbitrary ()
 
 spec :: Spec
-spec = pure ()
-
--- describe "Conway governance actions reordering" $ do
---   prop "preserves length when reordered" $
---     \actions -> Seq.length actions == Seq.length (reorderActions @Conway actions)
---   prop "sorts by priority" $
---     \actions ->
---       sort (toList (actionPriority . gasAction @Conway <$> actions))
---         == toList (actionPriority . gasAction <$> reorderActions actions)
---   prop "same priority actions are not rearranged" $
---     \a as ->
---       let filterPrio b = actionPriority (gasAction a) == actionPriority (gasAction b)
---        in filter filterPrio (toList $ reorderActions @Conway (a Seq.:<| as))
---             == filter filterPrio (toList $ reorderActions (a Seq.:<| as))
---   prop "orders actions correctly" $ do
---     actionsList <-
---       traverse
---         (>>= genGovActionStateFromAction)
---         (govActionGenerators @Conway)
---     let sortedActions = sortOn (actionPriority . gasAction) actionsList
---     shuffledActions <- shuffle actionsList
---     pure $
---       Seq.fromList sortedActions
---         == reorderActions (Seq.fromList shuffledActions)
+spec =
+  describe "Conway governance actions reordering" $ do
+    prop "preserves length when reordered" $
+      \(actions :: Seq.StrictSeq (GovActionState Conway)) ->
+        Seq.length actions `shouldBe` Seq.length (reorderActions @Conway actions)
+    prop "sorts by priority" $
+      \(actions :: Seq.StrictSeq (GovActionState Conway)) ->
+        sort (toList (actionPriority . gasAction @Conway <$> actions))
+          `shouldBe` toList (actionPriority . gasAction <$> reorderActions actions)
+    prop "same priority actions are not rearranged" $
+      \(a :: GovActionState Conway) (as :: Seq.StrictSeq (GovActionState Conway)) ->
+        let filterPrio b = actionPriority (gasAction a) == actionPriority (gasAction b)
+         in filter filterPrio (toList $ reorderActions @Conway (a Seq.:<| as))
+              `shouldBe` filter filterPrio (toList $ reorderActions (a Seq.:<| as))
+    prop "orders actions correctly" $
+      \(actionsList :: [GovActionState Conway]) -> do
+        let sortedActions = sortOn (actionPriority . gasAction) actionsList
+        Seq.fromList sortedActions `shouldBe` reorderActions (Seq.fromList actionsList)
```

Comment:
```suggestion
        let sortedActions = sortOn (actionPriority . gasAction) actionsList
        shuffledActions <- shuffle actionsList
        Seq.fromList sortedActions `shouldBe` reorderActions (Seq.fromList shuffledActions)
```

Reply:
I think the purpose of this test is to show that no matter how the actions are permuted, we still end up with the same output

Reply:
But I've implemented this in a weird way. I think that `sortedActions` should  just be replaced with `reorderActions`, since it's defined exactly as the actual implementation

Reply:
`shuffle` is a `Gen` monad here, while the test expects just an `IO`, after having moved the generator to the `Arbitrary` module. The generated list of actions is arbitrary and unsorted or in other words, shuffled already. I have checked this by eye-balling the generated list. I think this  should be fine as it is, since we already test that the sorting should result in the same operation as that of using `reorderActions`. What do you think?

Reply:
@Soupstraw do you still think we need a `shuffle` here in the test, or is it okay to get the shuffled stuff from the generator itself?

Reply:
We need to check that two different shuffles still give the same thing when reordered, so the shuffle is still necessary.

Reply:
@Soupstraw I would like to discuss this property with you when we can. 

I doubt if this property should hold, because the sorting is on a surjection (namely `actionPriority`): meaning that there are multiple elements in the list that will have the same `actionPriority` and so if we shuffle arbitrarily, we can still get a different order after sorting.

I don't know how this was working earlier though, I will check. And I'm not sure if I'm correct either. So let's discuss on Slack.

```haskell
data ShuffledGovActionStates era = 
  ShuffledGovActionStates [GovActionState era] [GovActionState era]
  deriving Show

instance 
  (Era era, Arbitrary (PParamsUpdate era)) => 
  Arbitrary (ShuffledGovActionStates era) 
  where
  arbitrary = do
    gass <- arbitrary
    shuffledGass <- shuffle gass
    pure $ ShuffledGovActionStates gass shuffledGass

-- [ ... ]

prop "orders actions correctly" $
  \(ShuffledGovActionStates @Conway gass shuffledGass) -> do
    let sortedGass = sortOn (actionPriority . gasAction) gass
    -- Set.fromList gass `shouldBe` Set.fromList shuffledGass
    -- Set.fromList sortedGass `shouldBe` Set.fromList shuffledGass
    Seq.fromList sortedGass `shouldBe` reorderActions (Seq.fromList shuffledGass)
```
This test fails.

Reply:
Ok, I went over the original test again and it looks like the idea behind this test is that each type of action in the list appears only once. You're right that this property wouldn't hold when we have more than one of some type of proposal.

Reply:
Sorry I missed that earlier @Soupstraw. Thank you! :raised_hands: